### PR TITLE
JENKINS-64069 Fix tables to div compat (minimum core 2.264 bump)

### PR DESCRIPTION
**Pull request by timja**
See https://issues.jenkins-ci.org/browse/JENKINS-64069?focusedCommentId=400785&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400785

For some reason jelly variables aren't being resolved correctly and I can't figure out why.

It work in this plugin when you run with `hpi:run` but if you do `mvn package -P quick-build` and then install it to a different Jenkins it doesn't work (name element isn't passed through)

I tried a few different options, a special tag just for this along with a local taglib and neither worked.

The local taglib didn't even pick up the `divBasedFormLayout`.

This bumps the minimum core to 2.264 (probably needs resubmitting by someone with write access to get a green build or changing the Jenkinsfile in a separate PR first.

If someone can figure out why the values aren't being passed through then we could avoid this PR but given this is the largest tables to divs breakage I think it would be good to just get this sorted.

@dwnusbaum @fqueiruga @oleg-nenashev

### CODE REVIEW

**Comment by dwnusbaum** [comment]
@timja so is 2.6.1 broken for everyone (in which case we should remove it from the update center ASAP), or only for users running Jenkins 2.264 or newer?

**Comment by timja** [comment]
> @timja so is 2.6.1 broken for everyone (in which case we should remove it from the update center ASAP), or only for users running Jenkins 2.264 or newer?

Unsure, needs testing.

**Comment by timja** [comment]
> @timja so is 2.6.1 broken for everyone (in which case we should remove it from the update center ASAP), or only for users running Jenkins 2.264 or newer?

I checked and it seems to be working fine and no need to remove from update center

**Review by dwnusbaum — APPROVED** [review]
I spent a few minutes trying to see if I could figure out what the problem was to be able to preserve support for older baselines, but I couldn't come up with anything, so I guess the only realistic option is to go ahead and update to 2.264 unless someone else has time to investigate further.

As far as users running 2.6.1 on old core versions, from a quick check it looks like there is a relatively significant UI display issue that should probably be fixed before we merge and release this unless we want to create a 2.6.1.x maintenance branch, see https://github.com/jenkinsci/branch-api-plugin/pull/226/files#r520138223.

**Comment by timja** [comment]
> I spent a few minutes trying to see if I could figure out what the problem was to be able to preserve support for older baselines, but I couldn't come up with anything, so I guess the only realistic option is to go ahead and update to 2.264 unless someone else has time to investigate further.
> 
> As far as users running 2.6.1 on old core versions, from a quick check it looks like there is a relatively significant UI display issue that should probably be fixed before we merge and release this unless we want to create a 2.6.1.x maintenance branch, see [#226 (files)](https://github.com/jenkinsci/branch-api-plugin/pull/226/files#r520138223).

we can just revert it

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,5 @@
 buildPlugin(configurations: [
         [ platform: "linux", jdk: "8", jenkins: null ],
-        [ platform: "linux", jdk: "11", jenkins: "2.249" ],
-        [ platform: "windows", jdk: "8", jenkins: "2.235.3", javaLevel: "8" ],
-        [ platform: "linux", jdk: "11", jenkins: "2.235.3", javaLevel: "8" ]
-    ])
+        [ platform: "linux", jdk: "11", jenkins: null ],
+        [ platform: "windows", jdk: "8", jenkins: null, javaLevel: "8" ],
```
Comment (jglick) [inline Jenkinsfile:4]: `jenkins` defaults to `null`, and `javaLevel` should be considered deprecated (no reason to ever use it with any plugin using a remotely recent POM).

```suggestion
        [ platform: "linux", jdk: "11" ],
        [ platform: "windows", jdk: "8" ],
```

File: src/main/resources/jenkins/branch/BranchSource/config.jelly [hunk 2]
```diff
@@ -22,19 +22,19 @@
  ~ THE SOFTWARE.
  -->
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:branch="/lib/branch-api">
-  <branch:rowWrapper name="source">
-    <branch:entryWrapper>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:branch="/lib/branch-api" xmlns:local="local">
+  <div class="tr" name="source">
+    <div>
       <input type="hidden" name="stapler-class" value="${sourceDescriptor.clazz.name}"/>
```
Comment (jglick) [inline src/main/resources/jenkins/branch/BranchSource/config.jelly:28]: OK for now but I have to wonder why this plugin is not using a standard databinding form control from Jenkins core that would work in any version.

**Comment by mpnico** [comment]
> @timja so is 2.6.1 broken for everyone (in which case we should remove it from the update center ASAP), or only for users running Jenkins 2.264 or newer?

UI is broken with 2.6.1 on jenkins LTS (2.249.3) for me. The UI is too small to be usable because of a missing colspan (which should be fixed by #231 )

**Review by oleg-nenashev — COMMENTED** [review]
Looks fine as an intermediate solution. Will become a big deal for plugin management tools which install `latest` without consulting with the LTS update centers

**Comment by dwnusbaum** [comment]
Rebuilding now that #229 was merged.

File: src/main/resources/jenkins/branch/BranchSource/config.jelly [hunk 3]
```diff
@@ -22,19 +22,19 @@
  ~ THE SOFTWARE.
  -->
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:branch="/lib/branch-api">
-  <branch:rowWrapper name="source">
-    <branch:entryWrapper colspan="3">
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:branch="/lib/branch-api" xmlns:local="local">
```
Comment (dwnusbaum) [inline src/main/resources/jenkins/branch/BranchSource/config.jelly:25]:
```suggestion
<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:branch="/lib/branch-api">
```

The `local` taglib can be removed, right?

**Comment by timja** [comment]
> Looks fine as an intermediate solution. Will become a big deal for plugin management tools which install `latest` without consulting with the LTS update centers

people can specify a version in that case or use plugin manager cli which looks it up properly...

**Comment by jglick** [comment]
> plugin management tools which install `latest` without consulting with the LTS update centers

:man_shrugging: user error.

**Comment by timja** [comment]
I can reproduce test failure locally, looking at it

**Comment by timja** [comment]
the `OrphanedItemStrategy` isn't being set properly on the folder.

Failure is probably, in another plugin, likely cloudbees-folder

File: pom.xml [hunk 4]
```diff
@@ -108,6 +108,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>cloudbees-folder</artifactId>
+      <version>6.15-rc617.b5ffac2bd6f5</version>
```
Comment (jglick) [inline pom.xml:111]:
```suggestion
      <version>6.15-rc617.b5ffac2bd6f5</version> <!-- TODO https://github.com/jenkinsci/cloudbees-folder-plugin/pull/165 -->
```

**Comment by timja** [comment]
Can you release with an incrementals test dependency? (Assume so but never tried)

**Comment by timja** [comment]
All those tests that previously fail pass with the incremental

**Comment by jglick** [comment]
> Can you release with an incrementals test dependency?

Pending [JENKINS-50804](https://issues.jenkins.io/browse/JENKINS-50804) nothing blocks you, but please do not.

**Comment by timja** [comment]
> > Can you release with an incrementals test dependency?
> 
> Pending [JENKINS-50804](https://issues.jenkins.io/browse/JENKINS-50804) nothing blocks you, but please do not.

Ok I guess after CI run complete we ignore the tests and revert the previous commit?

**Comment by dwnusbaum** [comment]
> Ok I guess after CI run complete we ignore the tests and revert the previous commit?

Can we just release `cloudbees-folder` first?

**Comment by timja** [comment]
> > Ok I guess after CI run complete we ignore the tests and revert the previous commit?
> 
> Can we just release `cloudbees-folder` first? It looks like the dependency is only in one direction.

@fcojfernandez ?

Yeah I guess that would work fine

**Comment by fcojfernandez** [comment]
@timja @dwnusbaum do you mean including https://github.com/jenkinsci/cloudbees-folder-plugin/pull/165?

**Comment by timja** [comment]
yes

**Comment by fcojfernandez** [comment]
cloudbees-folder-6.15 is out @dwnusbaum @timja

File: pom.xml [hunk 5]
```diff
@@ -108,6 +108,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>cloudbees-folder</artifactId>
+      <version>6.15-rc617.b5ffac2bd6f5</version> <!-- TODO https://github.com/jenkinsci/cloudbees-folder-plugin/pull/165 -->
```
Comment (dwnusbaum) [inline pom.xml:111]:
```suggestion
      <version>6.15</version>
```

**Comment by timja** [comment]
green build

**Review by dwnusbaum — APPROVED** [review]
@timja thanks for looking into this!

**Pull request merged**